torch/csrc/jit/autodiff.cpp

```diff
@@ -465,39 +465,49 @@ static value_list getReverseCaptures(Gradient& grad_desc) {
 // nodes we have in our graphs are simply constants, which are cheap to execute
 // and replicate, and so it's better to just copy them into the reverse graph,
 // without polluting the output lists unnecessarily.
-static void liftConstants(Gradient& grad_desc, ReverseDetails& rev_info) {
+static void liftConstants(Block* block, Block* if_not_in_this_block);
+
+static bool inBlock(Node* node, Block* container) {
+  Block* b = node->owningBlock();
+  while (b) {
+    if (b == container) {
+      return true;
+    }
+    b = b->owningNode() ? b->owningNode()->owningBlock() : nullptr;
+  }
+  return false;
+}
+
+static void liftConstants(Node* node, Block* if_not_in_this_block) {
   static const auto err = [](Value*) -> Value* {
     throw std::runtime_error("unexpected input");
   };
-  auto& graph = *grad_desc.f;
-  Block* reverse_block = rev_info.reverse_block;
-
-  for (Node* top_node : reverse_block->nodes()) {
-    AT_ASSERT(
-        top_node->kind() == prim::GradOf ||
-        top_node->kind() == prim::AutogradAdd ||
-        top_node->kind() == prim::AutogradZero);
-    if (top_node->kind() != prim::GradOf)
+  auto& graph = *node->owningGraph();
+  for (Value* input : node->inputs()) {
+    if (input->node()->kind() != prim::Constant)
       continue;
-    Block* grad_body = top_node->blocks().at(0);
-    for (Node* node : grad_body->nodes()) {
-      for (Value* input : node->inputs()) {
-        if (input->node()->kind() != prim::Constant)
-          continue;
-        if (input->node()->owningBlock() == grad_body)
-          continue;
-        Node* lifted_constant = graph.createClone(input->node(), err);
-        reverse_block->prependNode(lifted_constant);
-        GRAPH_DEBUG(
-            "Lifting constant ",
-            input->debugName(),
-            " from GradOf's block and adding ",
-            lifted_constant->output()->debugName(),
-            " to the backprop block");
-        node->replaceInputWith(input, lifted_constant->output());
-      }
-    }
+    if (inBlock(input->node(), if_not_in_this_block))
```

Comment:
Can you add a comment explaining what's going on here? It's hard for me to understand what this code does